Etg.Yams/Storage/IYamsRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Threading.Tasks;
+using Etg.Yams.Application;
+using Etg.Yams.Storage.Config;
+
+namespace Etg.Yams.Storage
+{
+    public interface IYamsRepository
```

Comment:
IApplicationRepository or IDeploymentRepository both sound good to me. Yams in front is redundant due to namespacing. I don't like the current one :)

Sent from my iPhone

> On Jan 23, 2016, at 9:42 AM, Nehme Bilal <notifications@github.com> wrote:
> 
> In Etg.Yams/Storage/IYamsRepository.cs:
> 
> > @@ -0,0 +1,16 @@
> > +﻿using System.Threading.Tasks;
> > +using Etg.Yams.Application;
> > +using Etg.Yams.Storage.Config;
> > +
> > +namespace Etg.Yams.Storage
> > +{
> > +    public interface IYamsRepository
> Few concerns about IYamsApplicationRepository:
> 
> A bit long
> Application is a bit redundant because it appears in the method names (e.g. UploadApplicationBinaries).
> Not clear whether it's the Repository for one Application or many (Applications).
> The repository contains more than application binaries (DeploymentConfig.json and maybe more in the future).
> Thoughts? @ReubenBond @galvesribeiro ?
> 
> Other possible names I can think of, IApplicationRepository, IApplicationsRepository, IYamsStorage, IDeploymentRepository, IDeploymentsRepository.
> 
> I like the current name though :)
> 
> —
> Reply to this email directly or view it on GitHub.
>